TICL Reorganisation

rovere opened the pull request:
#### PR description:

##### TICL Reorganization

Several changes entered this PR.
    
  * The data structures used by TICL have been moved into
      DataFormats/TICL, that has to be added as a CMSSW package.
      Tracksters and Tile are currently present. The encapsulation of
      std::array as a transient data member of a Tile is due to the fact
      that ROOT natively supports those and refuses to create ad-hoc
      dictionaries. Moreover, at present, an std::array cannot sit at the
      top of a ROOT branch requiring, once again, its encapsulation.
  * The "master Tile" has been added as a single producer. A unique Tile
      containing all the 2D reconstructed clusters is created once and put
      into the Event. All iterations will fetch it from the event and mask
      the 2D clusters on the fly. This required also some changes to the
      clusters filters logic. Now the filters produce also a vector<float>
      that represents the result of merging their input mask with their
      own selection. TICL iterations will make use of the filter's
      vector<float> to dynamically mask clusters from the "master Tile".
  * The fractions have been added to the Tracksters. The sharing of the
      energy of 2D clusters is flat and based on their multiplicity, i.e,
      a fraction of energy/multiplicity is assigned to all Tracksters that
      use the 2D clusters. Better splitting algorithms can be studies or
      Machine Learning techniques could be put at work here.
  * The hits and fraction of the Trackster are now correctly populated.
      This required an additional loop over the components of each 2D
      cluster. In the future the CaloCluster, PFCluster and hits and
      fractions data structures could be reviewed or, at least, TICL could
      use some other, more efficient, internal data representation and
      collapse into the legacy ones only at the very last stage.

#### PR validation:

`runTheMatrix.py -l limited -i all -j 4`

and other samples.

**Note to the reviewers**: all changes in `RecoLocalCalo/HGCalRecAlgos` are purely due to the code-format. No other changes have been committed there. I did so so that PR #26991 could be closed (if and when this PR gets merged)

tocheng committed:
update the way to tag bad ROCs

tocheng committed:
update the way to tag bad ROCs

tocheng committed:
remove unused variables

tocheng committed:
remove unused variables

makortel committed:
Change PluginFactory to return unique_ptr<Plugin>

tocheng committed:
Add description to SiPixelStatus producer

tocheng committed:
Add description to SiPixelStatus producer

tocheng committed:
Update ALCARECO production and harvest testers

leonardogiannini committed:
new features

leonardogiannini committed:
info builders and converter for DV

leonardogiannini committed:
changes int the global features

leonardogiannini committed:
class def change

leonardogiannini committed:
Deepvertex files and modifications

leonardogiannini committed:
TF producer for Deepvertex

leonardogiannini committed:
start cleaning from printout

leonardogiannini committed:
more cleaning

leonardogiannini committed:
more cleaning

leonardogiannini committed:
add comment

leonardogiannini committed:
switch off the deepvertex modules

leonardogiannini committed:
add configuration paramtere

deguio committed:
update to use the shape ID which is currently used in prod

deguio committed:
add realistic digitizer

deguio committed:
update HEback digitizer parameters

deguio committed:
scale signal based on the tile size

deguio committed:
scale signal based on the dose

deguio committed:
improvements and fixes

deguio committed:
fixes

deguio committed:
added validator for signal scaler analyzer

deguio committed:
add flags to enable features

deguio committed:
improvements

deguio committed:
better binning

deguio committed:
fix

deguio committed:
proper implementation of customization

deguio committed:
add noise computation from fluence params

leonardogiannini committed:
indentation and other fixes

leonardogiannini committed:
adjusted

leonardogiannini committed:
jet 4vec filler

leonardogiannini committed:
method in cc

leonardogiannini committed:
adjeusted

leonardogiannini committed:
recocand and minor changes

leonardogiannini committed:
minor changes

leonardogiannini committed:
added line for method moved to .cc

leonardogiannini committed:
no prefix

leonardogiannini committed:
xml and checksum

leonardogiannini committed:
weight file position

leonardogiannini committed:
changes for testing

leonardogiannini committed:
test cfg deepvertex

deguio committed:
prepare for PR

deguio committed:
replace cout with message logdebug

deguio committed:
fix

deguio committed:
fix code style

deguio committed:
propagate the noise_MIP config change

deguio committed:
optimizations and fixes

kandrosov committed:
Skeleton for DeepTau v2.

MRD2F committed:
Merge pull request #1 from kandrosov/deepTauId_v2_work

Skeleton for DeepTau v2

tocheng committed:
Add TH1F for debugging and make threshold configurable

Commit:
motifications for the TauBlockInputs

Commit:
fixed some issues

MRD2F committed:
Merge pull request #2 from MRD2F/deepTauId_v2_working

motifications for the TauBlockInputs

tocheng committed:
Using Tracker DQM framework instead of TH1 and TFileService

kandrosov committed:
Improved CellGrid implementation.

tocheng committed:
Fix bug in plot naming and add total Bpix and total Fpix plot

tocheng committed:
Still fill permanent bad components when run has low stat

tocheng committed:
Tune the overflow bin for dead ROCs

MRD2F committed:
Merge pull request #3 from kandrosov/deepTauId_v2_work

Improved CellGrid implementation.

Commit:
changes in the hadronBlock

Commit:
Merge remote-tracking branch 'my-cmssw/deepTauId_v2' into deepTauId_v2_working

tocheng committed:
Add a new threshold point 0.05

Commit:
updates in the hadron block

deguio committed:
new parametrization and geometry v10

Commit:
Fixed some problems with the hadron block

Commit:
Continue fixing some problems in the hadron block

Commit:
Continue fixing some problems in the hadron block

MRD2F committed:
Merge pull request #4 from MRD2F/deepTauId_v2_working

changes in the hadronBlock

tocheng committed:
Add more plots to check pixel quality

tocheng committed:
fix bug in calculating layer-1 occupancy

Commit:
Updates in the electron block

tocheng committed:
add per-lumi inst lumi from LHCinfo; add tree to host digi loss rate due to bad rocs

tocheng committed:
seperate PCL code and code to check thresholds

Commit:
Fix in the GetValue function

Commit:
Fixed some indentation problems

Commit:
Applied comments

tocheng committed:
fix bug in integer division and fix segmentation of assigning last payload

Commit:
applied comments

Commit:
Fixed indentation

MRD2F committed:
Merge pull request #5 from MRD2F/deepTauId_v2_working

Updates in the electron block

Commit:
improvements into the muon block

Commit:
Modifications in the muon block

Commit:
applied comments

MRD2F committed:
Merge pull request #6 from MRD2F/deepTauId_v2_working

Changes in the muon block

tocheng committed:
Add a boolean to turn off TFileService in case of emptyRun (to avoid segmentation fault)

Commit:
Modificacion to add v2 in the python config

Commit:
Fixed bug

Commit:
fixed bugs

Commit:
applied some comments

MRD2F committed:
Merge pull request #7 from MRD2F/deepTauId_v2_working

Modificacion to add v2 in the python config

tocheng committed:
different thresholds for different sub-components

MRD2F committed:
Fixed some bugs of previus PR

MRD2F committed:
Applied PR comments

MRD2F committed:
Fixed indentetion problem

MRD2F committed:
applied comments

MRD2F committed:
Merge pull request #8 from MRD2F/deepTauId_v2_working

 Fixed some bugs of previus PR

MRD2F committed:
Fixed some errors and create a new method printInputs

MRD2F committed:
return previus runDeepTauIDsOnMiniAOD.py configuration

MRD2F committed:
applied PR comments

MRD2F committed:
applied commets

MRD2F committed:
FIxed sample name

leonardogiannini committed:
classes version

leonardogiannini committed:
candidates

leonardogiannini committed:
remove comment

leonardogiannini committed:
range loop and more rewriting

leonardogiannini committed:
move to cc

leonardogiannini committed:
renaming

leonardogiannini committed:
move method to cc

MRD2F committed:
Merge pull request #9 from MRD2F/deepTauId_v2_working

 Fixed some errors and create a new method printInputs

MRD2F committed:
fixed some bugs in DeepTauId.cc and implemented new reading version function in runTauIdMVA.py

MRD2F committed:
remove some aditional spaces

MRD2F committed:
applied PR commets

MRD2F committed:
Merge pull request #10 from MRD2F/deepTauId_v2_working

Final PR

MRD2F committed:
Improved memory usage

mbluj committed:
Merge pull request #123 from MRD2F/deepTauId_v2

Integration of DeepTauId v2

mbluj committed:
Add DeepTauId 2017v2 to MiniAOD workflow

deguio committed:
implement comments from @perrotta

mbluj committed:
Initialization by a number rather than by a function

deguio committed:
update noise param in VFEProducer

mbluj committed:
Use `static constexpr` rather than `static const`

mbluj committed:
remove ambiguous enum in namespace

jbsauvan committed:
Fix propagation of noise value changes to VFE producer

mbluj committed:
remove unused variable

mbluj committed:
remove unused variable (bug fix)

kpedro88 committed:
isolate refs

guitargeek committed:
don't copy blocks, dtor for mualgo etc.

adewit committed:
Storing changes for running on CAF-Condor

tocheng committed:
remove the ntuple/TH1F/DQM plots that are used to test/check the threshold settings

tocheng committed:
update Era usage

tocheng committed:
remove un-needed commented lines

rovere committed:
Remove HGCGraph from within PatternRecognitionbyCA

rovere committed:
Move TICL customizations into python directory

rovere committed:
Add fractions in Tracksters

kandrosov committed:
Improved CPU timing of deepTauId.

- Split network into 3 parts: inner, outer and core.
- Execute inner and outer sub-networks only for non-empty cells.

rovere committed:
Code-format

kandrosov committed:
Bugfixes in DeepTauId.

rovere committed:
TICL Reorganization

Several changes entered this commit.

* The data structures used by TICL have been moved into
  DataFormats/TICL, that has to be added as a CMSSW package.
  Tracksters and Tile are currently present. The encapsulation of
  std::array as a transient data member of a Tile is due to the fact
  that ROOT natively supports those and refuses to create ad-hoc
  dictionaries. Moreover, at present, an std::array cannot sit at the
  top of a ROOT branch requiring, once again, its encapsulation.
* The "master Tile" has been added as a single producer. A unique Tile
  containing all the 2D reconstructed clusters is created once and put
  into the Event. All iterations will fetch it from the event and mask
  the 2D clusters on the fly. This required also some changes to the
  clusters filters logic. Now the filters produce also a vector<float>
  that represents the result of merging their input mask with their
  own selection. TICL iterations will make use of the filter's
  vector<float> to dynamically mask clusters from the "master Tile".
* The fractions have been added to the Tracksters. The sharing of the
  energy of 2D clusters is flat and based on their multiplicity, i.e,
  a fraction of energy/multiplicity is assigned to all Tracksters that
  use the 2D clusters. Better splitting algorithms can be studies or
  Machine Learning techniques could be put at work here.
* The hits and fraction of the Trackster are now correctly populated.
  This required an additional loop over the components of each 2D
  cluster. In the future the CaloCluster, PFCluster and hits and
  fractions data structures could be reviewed or, at least, TICL could
  use some other, more efficient, internal data representation and
  collapse into the legacy ones only at the very last stage.

deguio committed:
optimize radius computation

mbluj committed:
Merge pull request #126 from kandrosov/deepTau_2017v2_10_6_X_split

Improved CPU timing performance of deepTauId v2

clelange committed:
Remove lastLayerEE/FH magic numbers and use RecHitTools instead

slomeo committed:
Scenario 2021 with: Detailed Cavern, MB4 Neutron Shields (YB+2, YB-2, S 4 S13) and overlaps, inside the Muon System, fixed

leonardogiannini committed:
modified SeedingTrackInfoBuilder

leonardogiannini committed:
modified TrackPairInfoBuilder

leonardogiannini committed:
modified converter and producers - jet pt and eta cut - new methods in converter

leonardogiannini committed:
removed couts

Dr15Jones committed:
Modernized DetIdAssociatorESProducer

- removed framework constructs from DetIdAssociators
- created DetIdAssociatorMakers which get configurations from
  the ParameterSet, set what EventSetup data products will be
  consumed and then create the appropriate DetIdAssociator.

Dr15Jones committed:
Explicitly include header for DetIdAssociatorRecord

rovere committed:
Final round of code-format

rovere committed:
Add comments in the code

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
-code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10097

- This PR adds an extra 76KB to repository

Code check has found code style and quality issues which could be resolved by applying following patch(s)
- **code-checks**:
https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10097/code-checks.patch
e.g. `curl https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10097/code-checks.patch | patch -p1`
You can also run `scram build code-checks` to apply code checks directly

rovere committed:
Appy code-checks

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10098

- This PR adds an extra 80KB to repository

cmsbuild commented:
A new Pull Request was created by  @rovere (Marco Rovere) for master.

It involves the following packages:

DataFormats/TICL
RecoHGCal/TICL
RecoLocalCalo/HGCalRecAlgos


The following packages do not have a category, yet:

DataFormats/TICL
Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

@perrotta, @cmsbuild, @kpedro88, @slava77 can you please review it and eventually sign? Thanks.
@edjtscott, @vandreev11, @sethzenz, @felicepantaleo, @kpedro88, @argiro, @cseez, @pfs, @lgray this is something you requested to watch as well.
@davidlange6, @slava77, @fabiocos you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Commit:
Merged

slava77 commented:
@rovere 
your PR is missing clang-format changes in the following files

- RecoHGCal/TICL/plugins/HGCDoublet.cc
- RecoHGCal/TICL/plugins/HGCDoublet.h
- RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

Were these just missed in the commit file list or did your formatting tool somehow skipped these?
Please check and clarify.
Thank you.

Commit:
Push the other changes

kpedro88 commented on RecoLocalCalo/HGCalRecAlgos/interface/HGCalRecHitSimpleAlgo.h:
```diff
@@ -18,72 +18,67 @@
-      layer =  HcalDetId(baseid).depth() + 40;
-    } else if ( DetId::Forward == baseid.det() && HGCEE == baseid.subdetId() ) {
+    if (DetId::Hcal == baseid.det() && HcalEndcap == baseid.subdetId()) {
+      layer = HcalDetId(baseid).depth() + 40;
```
these hardcoded values should come from the geometry or RecHitTools

rovere replied:
Ciao @kpedro88,
as I said in the comment to this PR the package `RecoLocalCalo/HGCCalRecAlgos` has been added only to be able to include all `code-format` changes that were in #26991 
As such, I'd not consider this package as part of this review.

kpedro88 replied:
Okay, I've added this to #25472.

cmsbuild commented:
-1

Tested at: 28a481def3f2a07612f5ebe3230bcf08c2fea9dd

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/646/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals**

* **RelVals**:

When I ran the RelVals I found an error in the following workflows:
27434.0 step2
<pre>runTheMatrix-results/27434.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D35_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D35+RecoFullGlobal_2023D35+HARVESTFullGlobal_2023D35.log

</pre>29034.0 step2
<pre>runTheMatrix-results/29034.0_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D41+RecoFullGlobal_2023D41+HARVESTFullGlobal_2023D41/step2_TTbar_14TeV+TTbar_14TeV_TuneCUETP8M1_2023D41_GenSimHLBeamSpotFull14+DigiFullTrigger_2023D41+RecoFullGlobal_2023D41+HARVESTFullGlobal_2023D41.log

</pre>

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

kpedro88 commented:
This is an error from the IB, it should be fixed in the next IB.

leonardogiannini committed:
log with offset

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

rovere commented:
@slava77 I have no private tool at hand to perform code formatting.
I run
`scram b code-format`
and I hope this will touch and format all files in a package, not just the ones that have been modified. 
In any case, thanks for spotting this: I can manually run the clang formatter on those files.

rovere committed:
Run code-format on missed files

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10130

- This PR adds an extra 80KB to repository

cmsbuild commented:
Pull request #27026 was updated. @perrotta, @cmsbuild, @kpedro88, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 28a481def3f2a07612f5ebe3230bcf08c2fea9dd
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/679/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
+1
Tested at: 0bd89a01bb31b14dd86f630fd05e6a54d19d2b75
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/681/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/679/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3210678
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3210342
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/681/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3210678
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3210342
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

tocheng committed:
still upload one payload for low stat run

tocheng committed:
Revert "Add one payload to last_lumi+1 lumi section for the run : prompt payload = permanent bad components, other payload = empty

This reverts commit f12202dcbac027902a332230870126e358f9986c.

tocheng committed:
fix cmssw code check

tocheng committed:
fix the decription in SiPixelCalZeroBias producer

Dr15Jones committed:
Use enums for Entry type values

Declare all Entry types in the enum and use the values only from there.

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

adewit committed:
Making mps_scriptm.pl CC7-compatible

SenneVanPutte committed:
condor submission

makortel committed:
Remove ListOfRecordToMerge configuration parameter from SiPixelQualityESProducer as obsolete

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10180

- This PR adds an extra 108KB to repository

cmsbuild commented:
Pull request #27026 was updated. @perrotta, @cmsbuild, @kpedro88, @slava77 can you please check and sign again.

slava77 commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

Commit:
Reading a lost method call

cmsbuild commented:
+1
Tested at: 830a711b9c9316a91292d77f2bba640bc29bfdc5
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/741/summary.html

cmsbuild commented:
Comparison job queued.

kandrosov committed:
Applied comments from the central DeepTau v2 PR.

kandrosov committed:
Fixed deltaPhi calls.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/741/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 2 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3210222
 - DQMHistoTests: Total failures: 2
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3209886
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

Dr15Jones committed:
Switch to compile time structures for edm::Entry

Create constexpr data structures for all the name to code lookups.

slomeo committed:
deleted comments into the file

mbluj committed:
Merge pull request #128 from kandrosov/deepTau_2017v2_10_6_X_split

Applied comments from the central DeepTau v2 PR

leonardogiannini committed:
pass arguments as const

leonardogiannini committed:
optimize calls of IPs computations

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

slomeo committed:
added a missing line to cms.xml

makortel committed:
Fix use-after-move in CITKPFIsolationSumProducer

While the code worked with PluginFactory returning raw pointer, it
leads to segfault when it returns unique_ptr.

kpedro88 commented:
@rovere there is a conflict, please rebase

makortel committed:
Fix use-after-move in CITKPFIsolationSumProducerForPUPPI

rovere commented:
This is **really** disturbing. I guess that Sunanda's changes were made w/o formatting the code properly.
The changes in `RecoLocalCalo`, let me stress it once again, are not naturally part of this PR.
If the rebase turn out to be non-trivial, I'd simply remove all the changes there and leave the formatting for a later PR that someone else will make.
I'm not sure why code-formatting has not been requested on #27015 while it has been here.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10251

- This PR adds an extra 216KB to repository

cmsbuild commented:
Pull request #27026 was updated. @perrotta, @cmsbuild, @kpedro88, @slava77 can you please check and sign again.

perrotta commented:
@cmsbuild please test

cmsbuild commented:
The tests are being triggered in jenkins.

leonardogiannini committed:
build transient track outside the jet loop and store Abs IPs

leonardogiannini committed:
pass also ip2d

slava77 commented:
>  I'm not sure why code-formatting has not been requested on #27015 while it has been here.

the formatting (clang-format) is not required yet for the PRs to be tested.
It was not required for this PR either.

cmsbuild commented:
+1
Tested at: b20310ae0628d60fc4a6fdec078007354771e4f2
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/796/summary.html

cmsbuild commented:
Comparison job queued.

rovere commented:
@slava77  by the time #27015 has been merged, this PR with code-formatting in the packages touched by #27015 was already open.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ee013c/796/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 33
 - DQMHistoTests: Total histograms compared: 3215458
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 3215123
 - DQMHistoTests: Total skipped: 334
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 32 files compared)
 - Checked 137 log files, 14 edm output root files, 33 DQM output files

fabiocos commented:
@rovere the clang-format option was not yet enabled in code checks waiting for the bulk of the migration to be done, to minimize side effects. Unfortunately the more we advance the less this is true. 
Now the bulk migration is almost over, and we will start forcing the option in the code-checks.

slava77 commented on RecoHGCal/TICL/python/ticl_iterations.py:
```diff
@@ -19,6 +20,8 @@
 def TICL_iterations_withReco(process):
```
this file should eventually be replaced by parts that do not know about a process:
- a file with tasks defined to do the task
- a RecoHGCal/Configuration/python/RECOHGCal_EventContent_cff.py with all appropriate keep statements

slava77 reviewed (commented):
After looking around at the available DataFromats/ packages organization,
I think that the proposed content of DataFormats/TICL should be moved to DataFormats/HGCRecHit.
The Trackster as a DAG is a bit more generic and could perhaps be eventually moved to some more common place when it is used outside of the TICL.

Since we are moving things around, I'd like to question the location of the TICL algorithms. It is added in RecoHGCal as the only package in that subsystem. I missed this somehow.
 The algorithms introduced so far as a part of TICL are local to HGCAL (everything that they produce are parts of HGC clustering). The TICL package better be moved to the RecoLocalCalo subsystem.

cmsbuild committed:
Merge pull request #26966 from bsunanda/Run2-hcx206

Run2-hcx206 Take care of Phase2 situations for HCAL

cmsbuild committed:
Merge pull request #27107 from cms-sw/code-format-dqm-9c0ae1

Running code-format for dqm

cmsbuild committed:
Merge pull request #27104 from cms-sw/code-format-dqm-l1-9c6cd3

Running code-format for dqm-l1

cmsbuild committed:
Merge pull request #27081 from cms-sw/code-format-reconstruction-e2a068

Running code-format for reconstruction

cmsbuild committed:
Merge pull request #27109 from cms-sw/code-format-db-reconstruction-bdb1ee

Running code-format for db-reconstruction

cmsbuild committed:
Merge pull request #27103 from cms-sw/code-format-generators-10713a

Running code-format for generators

cmsbuild committed:
Merge pull request #26996 from clelange/clean_up_magic_numbers_master

Remove lastLayerEE/FH magic numbers and use RecHitTools instead

cmsbuild committed:
Merge pull request #27004 from slomeo/MuonOverlapsFixBis

Scenario 2021 with: Detailed Cavern, MB4 Neutron Shields (YB+2, YB-2,…

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-tidy

cmsbuild committed:
Clang-Format

cmsbuild committed:
Clang-Format

cmsbuild committed:
Merge pull request #27108 from cms-sw/code-format-analysis-reconstruction-xpog-2c767b

Running code-format for analysis-reconstruction-xpog

cmsbuild committed:
Merge pull request #27105 from cms-sw/code-format-analysis-simulation-61ab85

Running code-format for analysis-simulation

cmsbuild committed:
Merge pull request #27112 from cms-sw/code-format-alca-reconstruction-95ff1f

Running code-format for alca-reconstruction

fabiocos committed:
Post clang-format fixes according to review comments

slava77 commented:
uhm, it looks like something else from RecoLocalCalo/HGCalRecAlgos was merged (was it #26996 ?).
Marco, you will need to rebase, sorry.
It was a nice idea to put the code formatting in this PR, but it apparently coincided with a lot of other HGCAL reco changes in other PRs.

slava77 commented:
If I'm not mistaken, we currently have #27049, #27128 and #27129 overlapping with this PR (in addition to three more that were already merged in the last few days; two rebases already made and one more expected).

cmsbuild committed:
Merge pull request #26088 from leonardogiannini/deepvertex105x

First version of DeepVertex b-tagger

cmsbuild committed:
Merge pull request #27010 from Dr15Jones/consumesDetIdAssociatorESProducer

Modernized DetIdAssociatorESProducer

cmsbuild committed:
Merge pull request #27050 from tocheng/SiPixelQualityForUL_final

SiPixel quality harvester for Ultra legacy

cmsbuild committed:
Merge pull request #27087 from makortel/pixelQualityListOfRecordToMerge

Remove ListOfRecordToMerge configuration parameter from SiPixelQualityESProducer

cmsbuild committed:
Merge pull request #27092 from SenneVanPutte/condor-submission-PR

TkAlignment HipPy condor submission

cmsbuild committed:
Merge pull request #27093 from andrius-k/dqm-binbybin-string-fix2

Returning a lost method call in DQM bin by bin comparison tool

cmsbuild committed:
Merge pull request #27111 from cms-sw/code-format-analysis-reconstruction-864996

Running code-format for analysis-reconstruction

cmsbuild committed:
Merge pull request #27113 from cms-sw/code-format-reconstruction-e1f636

Running code-format for reconstruction

cmsbuild committed:
Merge pull request #27127 from Dr15Jones/betterPSetEntry

Improvements to PSet Entry implementation

cmsbuild committed:
Merge pull request #27133 from cms-sw/code-format-geometry-3d1376

Running code-format for geometry

cmsbuild committed:
Merge pull request #27139 from cms-sw/code-format-dqm-a78e9b

Running code-format for dqm

cmsbuild committed:
Merge pull request #27140 from cms-sw/code-format-reconstruction-b49911

Running code-format for reconstruction

cmsbuild committed:
Merge pull request #27118 from makortel/pluginFactoryCITKPFIsolationSumProducer

Fix use-after-move in CITKPFIsolationSumProducer(ForPUPPI)

cmsbuild committed:
Merge pull request #26796 from cms-tau-pog/CMSSW_10_6_X_tau-pog_DeepTau2017v2

Update of DeepTauID to ver. 2017v2

cmsbuild committed:
Merge pull request #27143 from fabiocos/fc-fixFormat20190607

Code format fixes after CMSSW_11_0_X_2019-06-06-2300

cmsbuild committed:
Merge pull request #26864 from guitargeek/remove_egamma_pf_legacy_11X_followup

Follow up on "Remove PFElectronAlgo and PFPhotonAlgo used back in Run 1"

cmsbuild committed:
Merge pull request #27102 from cms-sw/code-format-analysis-db-6c639b

Running code-format for analysis-db

cmsbuild committed:
Merge pull request #27134 from cms-sw/code-format-reconstruction-upgrade-4c8c8f

Running code-format for reconstruction-upgrade

cmsbuild committed:
Merge pull request #27136 from cms-sw/code-format-l1-upgrade-a337f4

Running code-format for l1-upgrade

cmsbuild committed:
Merge pull request #27138 from cms-sw/code-format-alca-db-233572

Running code-format for alca-db

fabiocos commented:
This is what I can see:
```
Testing PR #  27026  for branch  master
File  RecoLocalCalo/HGCalRecAlgos/interface/HGCal3DClustering.h  modified in PRs #  ['27049', '27128', '27026']
File  RecoLocalCalo/HGCalRecAlgos/interface/HGCalRecHitSimpleAlgo.h  modified in PRs #  ['27129', '27026']
File  RecoLocalCalo/HGCalRecAlgos/interface/KDTreeLinkerAlgoT.h  modified in PRs #  ['27049', '27026']
File  RecoLocalCalo/HGCalRecAlgos/interface/KDTreeLinkerToolsT.h  modified in PRs #  ['27049', '27026']
File  RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h  modified in PRs #  ['27128', '27026']
File  RecoLocalCalo/HGCalRecAlgos/src/HGCal3DClustering.cc  modified in PRs #  ['27049', '27026']
File  RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc  modified in PRs #  ['27128', '27026']
```

cmsbuild committed:
Merge pull request #27141 from cms-sw/code-format-alca-9c25db

Running code-format for alca

cmsbuild committed:
Merge pull request #27135 from cms-sw/code-format-simulation-f7b8d3

Running code-format for simulation

cmsbuild committed:
Merge pull request #27137 from cms-sw/code-format-simulation-upgrade-b74f46

Running code-format for simulation-upgrade

cmsbuild committed:
Merge pull request #26811 from deguio/HGCHEbackDigitizer

improvements to the HGCHEback digitizer and implementation of a end-of-life scenario for the HGC scintillator section

cmsbuild committed:
Merge pull request #27097 from makortel/pluginFactoryUniquePtr

Change PluginFactory to return unique_ptr

cmsbuild committed:
Merge pull request #27144 from adewit/from-CMSSW_11_0_X_2019-06-05-2300

TkAl MillePede: Make condor submission default

rovere committed:
Fix conflicts after merging of #26996

rovere committed:
Run code format on changes introduced by #26996

rovere committed:
Fix bug in merging

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-27026/10307

- This PR adds an extra 100KB to repository

- Found files with invalid states:
  - RecoTauTag/RecoTau/plugins/MuonHitMatch.cc:
    - Added: ff7663e5f7d7bf7578c47af5a7677488be76f7dc
    - Deleted: e86cb9f571f4e73f8093298f17423b64f2cb34c2
  - RecoTauTag/RecoTau/interface/MuonHitMatch.h:
    - Added: ff7663e5f7d7bf7578c47af5a7677488be76f7dc
    - Deleted: e86cb9f571f4e73f8093298f17423b64f2cb34c2
  - SimCalorimetry/HGCalSimProducers/data/doseParams_3000fb.txt:
    - Added: 27a241f9b39ce99608af3fb09860703567e411a7
    - Modified: dfa36e3a0953b390809af2869f8bede4799ba67e, c38ae091ed84bc6ada9982df7616767f1a1045ae
    - Deleted: 7730e1625fee9890b86dd670d5b1305ff3a1b9f4

cmsbuild commented:
Pull request #27026 was updated. @andrius-k, @schneiml, @ianna, @kpedro88, @rekovic, @fioriNTU, @tlampen, @alberto-sanchez, @pohsun, @santocch, @peruzzim, @perrotta, @civanch, @cmsbuild, @agrohsje, @smuzaffar, @Dr15Jones, @cvuosalo, @efeyazgan, @mdhildreth, @jfernan2, @tocheng, @slava77, @ggovi, @qliphy, @benkrikler, @kmaeshima, @christopheralanwest, @franzoni, @fgolf can you please check and sign again.
